src/id/shinigami/src/eu/kanade/tachiyomi/extension/id/shinigami/Shinigami.kt

```diff
@@ -83,70 +56,195 @@ class Shinigami : Madara("Shinigami", "https://shinigami09.com", "id"), Configur
-            ?: throw Exception("chapter_data script not found")
+        return GET(url, apiHeaders)
+    }
 
-        val deobfuscated = Deobfuscator.deobfuscateScript(script)
-            ?: throw Exception("Unable to deobfuscate chapter_data script")
+    override fun latestUpdatesParse(response: Response): MangasPage = popularMangaParse(response)
 
-        val keyMatch = KEY_REGEX.find(deobfuscated)?.groupValues
-            ?: throw Exception("Unable to find key")
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$apiUrl/v1/manga/list".toHttpUrl().newBuilder()
+            .addQueryParameter("page", page.toString())
+            .addQueryParameter("page_size", "30")
 
-        val chapterData = json.decodeFromString<CDT>(
-            CHAPTER_DATA_REGEX.find(script)?.groupValues?.get(1) ?: throw Exception("Unable to get chapter data"),
-        )
-        val postId = POST_ID_REGEX.find(script)?.groupValues?.get(1) ?: throw Exception("Unable to get post_id")
-        val otherId = OTHER_ID_REGEX.findAll(script).firstOrNull { it.groupValues[1] != "post" }?.groupValues?.get(2) ?: throw Exception("Unable to get other id")
-        val key = otherId + keyMatch[1] + postId + keyMatch[2] + postId
-        val salt = chapterData.s.decodeHex()
+        if (query.isNotEmpty()) {
+            url.addQueryParameter("q", query)
+        }
 
-        val unsaltedCiphertext = Base64.decode(chapterData.ct, Base64.DEFAULT)
-        val ciphertext = salted + salt + unsaltedCiphertext
+        // TODO: search by tag/genre/status/etc
 
-        val decrypted = CryptoAES.decrypt(Base64.encodeToString(ciphertext, Base64.DEFAULT), key)
-        val data = json.decodeFromString<List<String>>(decrypted)
-        return data.mapIndexed { idx, it ->
-            Page(idx, document.location(), it)
+        return GET(url.build(), apiHeaders)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage = popularMangaParse(response)
+
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl/series/${manga.url}"
+    }
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        // Migration from old web urls to the new api based
+        if (manga.url.startsWith("/series/")) {
+            throw Exception("Migrate dari $name ke $name (ekstensi yang sama)")
         }
+
+        return GET("$apiUrl/v1/manga/detail/${manga.url}", apiHeaders)
     }
 
-    private fun randomString(length: Int): String {
-        val charPool = ('a'..'z') + ('A'..'Z')
-        return List(length) { charPool.random() }.joinToString("")
+    override fun mangaDetailsParse(response: Response): SManga {
+        val mangaDetailsResponse = response.parseAs<ShinigamiMangaDetailDto>()
+        val mangaDetails = mangaDetailsResponse.data
+
+        return SManga.create().apply {
+            author = mangaDetails.taxonomy["Author"]?.joinToString(", ") { it.name }.orEmpty()
```

Comment:
`, ` is the default and can be omitted:
```suggestion
            author = mangaDetails.taxonomy["Author"]?.joinToString { it.name }.orEmpty()
```